### DISCUSSION TITLE
[Feature] 360 video file support

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [immich-app/immich#5130] [Feature Request]
### The feature

I have a GoPro max 360 camera for 360 degree videos and pictures. Immich displays the 360 pictures perfectly - allowing rotation. But it support any *.360 video files (its as if it does not recognize them as valid video files). 

Note: All of my originals are stored on a NAS and I add them to Immich as a library.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by mchiappinam** [comment]
I'm not able to upload .360 videos through the browser. Are you able to do it somehow? Or you're talking just about visualizing those videos locally through the Immich app?

**Comment by JesseWebDotCom** [comment]
I only use libraries, no browser uploading for me (so immich is re-encoding video from my NAS)

**Reply by mchiappinam** [reply]
Got it

**Reply by mchiappinam** [reply]
Are you sure Immich is processing the .360 video and not the .mp4 that Quik generate when saving the non-edited video to photos with the option "Save to Photos"? 

I'm asking that because I accidentally generated that .mp4 10 minutes ago and I was trying to figure out where Immich saved the original video, however that was actually the original video.

After that, I did tried to upload the .360 video to the library/admin/today's.../ but I don't see it on the library through the app or browser, I think Immich doesn't list pictures from the folder, I think it does list from the database and then decide if will show the thumbnails or not. (I might be wrong, I'm just guessing)

**Reply by JesseWebDotCom** [reply]
I don't use Quick - pull directly from the card to my NAS. But also my Immich instance is wiped right now as I'm debugging an HEVC issue (so I can't confirm further).

**Reply by JesseWebDotCom** [reply]
I think you're correct - I think Immich is not encoding any *.360 files. I will update this issue.

**Reply by JesseWebDotCom** [reply]
@mchiappinam - I kept refocused this feature request to be around adding support for video files with .360 file extensions. I opened this issue on immich flattening 360 video (ex. .mp4): https://github.com/immich-app/immich/issues/5414

**Reply by mchiappinam** [reply]
Ohh after reading #5414 i got it. Hopefully that gets implement soon!

**Reply by shtse8** [reply]
half year later, the issue still here.

**Reply by thost96** [reply]
any news on this?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by schulzalxs** [comment]
I would also really appreciate this feature!

**Comment by Sawtaytoes** [comment]
I ran into this today with `insv` video from Insta360 cameras. I was on a trip, and the ATV we drove had these 360 videos (2 video tracks and 1 audio track). It'd be nice if I had a way of playing these correctly in Immich (same as the original poster). Currently, it only shows one of the two video tracks (the first one).

**Reply by schulzalxs** [reply]
I ran into the same thing with Insta360 .insv files. The camera / rig records dual-fisheye 360 footage plus gyro/meta data in Insta360’s proprietary .insv container. Immich can ingest .insv (they are basically MP4 containers), but it doesn’t understand the Insta360-specific stuff inside – so ffmpeg / the player just picks the first video stream, which is why we only see one half / one lens.

To get a real 360° experience, the footage has to be stitched and projected to an equirectangular 2:1 MP4 with proper 360 metadata (the same format that YouTube / Google VR expect). That stitching + stabilization step is what Insta360 Studio / the mobile app do. It’s non-trivial to do this generically inside Immich (proprietary lens model, gyro-based stabilization, heavy CPU/GPU load, licensing of the Insta360 SDK), so right now Immich treats .insv as “just another MP4”.

My workaround:
	•	import the .insv into Insta360 Studio / the Insta360 app
	•	export a stitched 360° MP4 (equirectangular, H.264/HEVC)
	•	import that MP4 into Immich

If Immich ever wants to support Insta360 natively, it would basically need a stitching pipeline (dual-fisheye → equirectangular) and to preserve / re-write the 360 metadata when transcoding. Until then, converting via Insta360 Studio first is probably the only reliable way.

**Reply by jagjordi** [reply]
You might want to check this little thing I have been working on, its far from complete but it helps a bit in managing large libraries of insta360 videos: https://github.com/jagjordi/insta360-autostitcher

**Comment by BoPeng** [comment]
+1 so that i can be notified with the progress. Hopefully insta360 can open source its video playing engine in the near future.

**Reply by bo0tzz** [reply]
Leaving a comment notifies everyone unnecessarily; if you want to subscribe, please use the button in the sidebar instead.

**Comment by daallgeier** [comment]
Running into this as well. I have X3 and X5 footage. X5's footage gladly comes as a one-piece and is able to be previewed. 

Both come with a pre-encoded low-res version (.lrv) from the SD card.  Which would reduce workloads on the server to generate another set of low res video. Just **copy\*** and link that lrv file (and have that copy reside on the SSD/cache dataset). 

My wish would be to a full preview with panning and zooming on the web and mobile. Currently it is just a circle and no movement possible.  See at the end. 

*copy not move because the lrv version should stay in that location for easy editing access with the Insta360 studio. 

X3 footage usually comes named as: 
```
VID_20231129_224118_00_013.insv     # _00_ outward facing camera
VID_20231129_224118_10_013.insv*    # _10_ inward facing camera
LRV_20231129_224118_11_013.lrv*     # _11_ low res version (360) 
```

X4 and later footage usually comes as:
```
LRV_20251113_173915_01_146.lrv*      # _01_ low res version (360) 
VID_20251113_173915_00_146.insv*     # _00_ full 5k version (360)
```

although i am renaming them via a script to something like this to make sure lrv files and actual files are next to each other in the file picker. 
 ```
    20231129_224118_013_LRV_11.lrv
    20231129_224118_013_VID_00.insv
    20231129_224118_013_VID_10.insv
    20231130_214247_055_IMG_00.insp
    20251113_173915_146_LRV_01.lrv
    20251113_173915_146_VID_00.insv
```

preview (not trying to brag) 
<img width="2297" height="1404" alt="CleanShot 2026-01-27 at 17 59 49" src="https://github.com/user-attachments/assets/d4173ef2-aeeb-45b8-a2b8-bd625f8589dd" />

**Reply by jagjordi** [reply]
If you just want to get the alread-stitched, low-res video displayed as a real 360 you  just need to set the metadata tags. 
I have had success using this fork of spatialmedia: https://github.com/timminator/Spatial-Media-Metadata-Injector.git
However if you want to get the full resolution one, first it needs to be stitched into 360.  You can do it manually with the phone app, the desktop video editor. 
If you want to get it somehow automated you might be interested in checking this: https://github.com/jagjordi/insta360-autostitcher

**Reply by daallgeier** [reply]
so i have not spent any time on this since.  

now that i have moved my transcoding workload off of the nas and onto a mac mini i am running into a problem with the file sizes and the re-encoding workload.  

as we do already have low res version s (.lrv) i would want these to be run as the transcoded version.  i am dealing with about 12TiB of data here so there is a huge cost to retranscode everything and transcode in the future.  i am reading that this is not already possible but i want to stake my interest here

**Comment by marekStef** [comment]
any news to this

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/
